### ISSUE TITLE
Remove a 2.4 compatible use of the warnings module in deprecate

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5865]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)| @Julian reported|
|-|-|
|Trac ID|trac#5865|
|Type|enhancement|
|Created|2012-08-14 15:10:57Z|

Attachments:

* [deprecate-24.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f40/f4019dd83f93e8c12168c043b54062d8a4cfe6de/69a6cb9a31ea7f8e7ec62a5fc31418ba5f9da89f.patch) (457 bytes) - added by Julian on 2012-08-14 15:11:08Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5865 5865
type__enhancement enhancement
reporter__Julian Julian
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1344957057000000 1344957057000000
changetime__1347207606000000 1347207606000000
version__None None
owner__therve therve

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve set owner to @therve|
|-|-|

Looks good!

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set status to `closed`|
|-|-|

(In [35561]) Remove a 2.4 compatible use of the warnings module in t.p.deprecate

Author: Julian
Reviewer: therve
Fixes: [#5865](https://github.com/twisted/twisted/issues/5865)
